src/Kmd.Logic.CitizenDocuments.Client/CitizenDocumentsClient.cs

```diff
@@ -89,6 +94,144 @@ public async Task<CitizenDocumentUploadResponse> UploadAttachmentWithHttpMessage
             }
         }
 
+        /// <summary>
+        /// Uploads the single citizen document.
+        /// </summary>
+        /// <param name="document">Original citizen document.</param>
+        /// <param name="citizenDocumentUploadRequestModel">citizenDocumentUploadRequestModel to update to db.</param>
+        /// <returns>The fileaccess page details or error if isn't valid.</returns>
+        /// <exception cref="ValidationException">Missing cpr number.</exception>
+        /// <exception cref="SerializationException">Unable to process the service response.</exception>
+        /// <exception cref="LogicTokenProviderException">Unable to issue an authorization token.</exception>
+        /// <exception cref="CitizenDocumentsException">Invalid Citizen document configuration details.</exception>
+        public async Task<CitizenDocumentUploadResponse> UploadFileAttachmentDirectlyToStorageAsync(Stream document, CitizenDocumentUploadRequestModel citizenDocumentUploadRequestModel)
+        {
+            if (document == null)
+            {
+                throw new ArgumentNullException(nameof(document));
+            }
+
+            if (citizenDocumentUploadRequestModel == null)
+            {
+                throw new ArgumentNullException(nameof(citizenDocumentUploadRequestModel));
+            }
+
+            if (citizenDocumentUploadRequestModel.CitizenDocumentConfigId == null)
+            {
+                throw new Exception("Citizen Document Configuration is null");
+            }
+
+            if (citizenDocumentUploadRequestModel.Cpr == null)
+            {
+                throw new Exception("Cpr is null");
+            }
+
+            if (citizenDocumentUploadRequestModel.DocumentName == null)
+            {
+                throw new Exception("DocumentName is null");
+            }
+
+            if (citizenDocumentUploadRequestModel.RetentionPeriodInDays == null)
+            {
+                throw new Exception("RetentionPeriodInDays is null");
+            }
+
+            if (citizenDocumentUploadRequestModel.DocumentType == null)
+            {
+                throw new Exception("DocumentType is null");
+            }
+
+            var client = this.CreateClient();
+            var documentId = Guid.NewGuid();
+            var storageDocName = citizenDocumentUploadRequestModel.DocumentName.Trim().Replace(".", string.Empty) + "_" + documentId + ".pdf";
+            var responseSasUri = await client.StorageAccessWithHttpMessagesAsync(
+                                   subscriptionId: new Guid(this.options.SubscriptionId),
+                                   documentName: storageDocName).ConfigureAwait(false);
+
+            if (responseSasUri.Response.StatusCode != System.Net.HttpStatusCode.OK)
+            {
+                throw new Exception("You don't have permission to upload");
```

Comment:
`Exception` is base type and it should never be throws. Please use more specific type for exception.

Reply:
Changed Exception